base/src/commonMain/kotlin/monitor/ServiceMonitor.kt

```diff
@@ -0,0 +1,46 @@
+/*
+ Copyright 2021 Splendo Consulting B.V. The Netherlands
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+
+package com.splendo.kaluga.base
+
+import com.splendo.kaluga.logging.debug
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+
+abstract class ServiceMonitor {
+
+    protected val TAG: String = this::class.simpleName ?: "ServiceMonitor"
+
+    abstract val isServiceEnabled: Boolean
+
+    protected val _isEnabled = MutableStateFlow(isServiceEnabled)
+    val isEnabled = _isEnabled.asStateFlow()
+
+    open fun startMonitoring() {
+        debug(TAG) { "Start monitoring service state ($isEnabled)" }
```

Comment:
```suggestion
        debug(TAG) { "Start monitoring service state ($isServiceEnabled)" }
```

Otherwise it will print Flow info

Reply:
3984652